go-dlv.el

```diff
@@ -143,27 +143,34 @@ and source-file directory for your debugger."
 (defun dlv-current-func ()
   "Debug the current program or test stopping at the beginning of the current function."
   (interactive)
-  (let (current-test-name current-func-loc)
+  (let (current-test-name current-bench-name current-func-loc)
     ;; find the location of the current function and (if it is a test function) its name
     (save-excursion
       (when (go-beginning-of-defun)
         (setq current-func-loc (format "%s:%d" buffer-file-name (line-number-at-pos)))
-        ;; if we are looking at the test function populate current-test-name
+        ;; Check for Test or Benchmark function, set current-test-name/current-bench-name
         (when (looking-at go-func-regexp)
           (let ((func-name (match-string 1)))
-            (when (and (string-match-p "_test\.go$" buffer-file-name)
-                       (string-match-p "^Test\\|^Example" func-name))
-              (setq current-test-name func-name))))))
+            (when (string-match-p "_test\.go$" buffer-file-name)
+              (cond
+               ((string-match-p "^Test\\|^Example" func-name)
+                (setq current-test-name func-name))
+               ((string-match-p "^Benchmark" func-name)
+                (setq current-bench-name func-name))))))))
 
     (if current-func-loc
         (let (gud-buffer-name dlv-command)
-          (if current-test-name
-              (progn
-                (setq gud-buffer-name "*gud-test*")
-                (setq dlv-command (concat go-dlv-command-name " test -- -test.run " current-test-name)))
-            (progn
-              (setq gud-buffer-name "*gud-debug*")
-              (setq dlv-command (concat go-dlv-command-name " debug"))))
+          (cond
+           (current-test-name
+            (setq gud-buffer-name "*gud-test*")
+            (setq dlv-command (concat go-dlv-command-name " test -- -test.run " current-test-name)))
+           (current-bench-name
+            (setq gud-buffer-name "*gud-test*")
+            (setq dlv-command (concat go-dlv-command-name " test -- -test.run '^$' -test.bench " current-bench-name)))
+
+           (nil
```

Comment:
did you push an old commit? see https://github.com/benma/go-dlv.el/pull/17#discussion_r301652319

Reply:
One of those days...

Reply:
I guess so, because now the `^$` is gone again :rofl:

Reply:
Sigh! OK, I think I did it right this time. As penance I even verified that it skips this pathological test:
```
func TestBad(t *testing.T) {
	t.Run("", func(t *testing.T) { t.Fatal("FIXME") })
}
```
And it does - the inner test name is "TestBad/#00", not "" :)

BTW in case I didn't mention it before - dlv-current-func is a a thoroughly brilliant feature. Never before could I just plant my cursor and debut a test! Brilliant!

Reply:
Excellent!